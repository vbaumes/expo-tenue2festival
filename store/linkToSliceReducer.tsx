import { createSlice } from '@reduxjs/toolkit';

let initialState: { category: string; gender: string; product: Object } = {
    category: '',
    gender: '',
    product: {}
};

export const linkToSlice = createSlice({
  name: 'linkTo',
  initialState,
  reducers: {
    set_category: (state, action) => {
      return { ...state, category: action.payload };
    },
    set_gender: (state, action) => {
      return { ...state, gender: action.payload };
    },
    set_product: (state, action) => {
      return { ...state, product: action.payload}
    }
  },
});
export const { set_category, set_gender, set_product } = linkToSlice.actions;
export const linkToSliceReducer = linkToSlice.reducer;