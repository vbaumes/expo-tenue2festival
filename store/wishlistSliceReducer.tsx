import { createSlice } from '@reduxjs/toolkit';

let initialState: { products: Array<Object> } = {
    products: []
};

export const wishlistSlice = createSlice({
  name: 'wishlist',
  initialState,
  reducers: {
    set_products: (state, action) => {
      return { ...state, products: action.payload };
    },
  },
});
export const { set_products } = wishlistSlice.actions;
export const wishlistSliceReducer = wishlistSlice.reducer;