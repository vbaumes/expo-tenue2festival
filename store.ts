import { configureStore } from '@reduxjs/toolkit';
import { linkToSliceReducer } from './store/linkToSliceReducer';

export const store = configureStore({
  reducer: {
    linkTo: linkToSliceReducer
  },
  middleware: (getDefaultMiddleware) => 
    getDefaultMiddleware({
      immutableCheck: false,
      serializableCheck: false
    })
})

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch