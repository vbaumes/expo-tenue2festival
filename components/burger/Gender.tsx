import React from 'react';
import {Text, StyleSheet, TouchableOpacity} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from '../../store';
import { set_gender } from '../../store/linkToSliceReducer';

export type Props = {
    name: string,
    id: number
};

const Gender: React.FC<Props> = ({name, id}) => {
  const dispatch = useDispatch();
  let genderStyle = useSelector((state: RootState) => {
    return (state.linkTo.gender === id) ? styles.isSelected : styles.notSelected;
  })
  
  return (
    <TouchableOpacity style={genderStyle} onPress={() => dispatch(set_gender(id))}>
      <Text>{name}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  notSelected: {
    backgroundColor: '#CECECE',
    color: '#717171',
    height: 70,
    width: 200,
    borderColor: 'black',
    borderWidth: 1,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  isSelected: {
    backgroundColor: '#717171',
    height: 70,
    width: 200,
    margin: 'auto',
    borderColor: 'black',
    borderWidth: 1,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default Gender;