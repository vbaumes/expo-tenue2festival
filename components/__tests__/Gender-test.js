import * as React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../store';

import Gender from '../burger/Gender';

it(`renders correctly`, () => {
  const tree = renderer.create(<Provider store={store}><Gender name="VÃªtements" /></Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});
