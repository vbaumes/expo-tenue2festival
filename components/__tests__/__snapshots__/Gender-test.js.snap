// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderColor": "black",
      "borderWidth": 1,
      "display": "flex",
      "height": 75,
      "justifyContent": "center",
      "margin": "auto",
      "opacity": 1,
      "width": 200,
    }
  }
>
  <Text>
    VÃªtements
  </Text>
</View>
`;
